soroban-env-host/src/host.rs

```diff
@@ -467,6 +467,17 @@ impl Host {
         self.0.budget.charge(ty, input)
     }
 
+    pub(crate) fn with_debug_budget<T, F, E>(&self, f: F, e: E) -> T
```

Comment:
I think a more appropriate name here would actually be something like `with_debug_mode` or something like that, as it also bypasses fn when we're not in debug mode (which is really convenient but should probably be reflected in the name).

Reply:
Not so sure about that. This is a wrapping of "asserting we are in debug mode" and "use the internal budget for metering", so feels like `with_debug_budget` is the most accurate term. Or putting it another way, would there be any scenario where `DEBUG` mode and using internal budget diverges (we want former but don't want to do the latter), if no such scenario exists, then I think `with_debug_mode` also makes sense.

Reply:
I think `with_debug_mode` reflects the use case better - I don't think we want to do any debug/diagnostic calls without debug budget (and vice versa). At least for now I can't think of such scenarios (but I could be missing something of course).

Reply:
Renamed to `with_debug_mode`.